data/transactions/logic/eval_test.go

```diff
@@ -5691,3 +5691,164 @@ int 88
 switch done1 done2; done1: ; done2: ;
 `, 8)
 }
+
+func TestMatch(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	// take the 0th label with int cases
+	testAccepts(t, `
+int 0
+int 1
+int 0
+match zero one
```

Comment:
Let's do these test with big weird numbers like 37 and 62 or some such, so it's obvious at a glance that we're not talking about indices into the labels.